pkg/capi/aws.go

```diff
@@ -235,6 +237,91 @@ var _ = Describe("Cluster API AWS MachineSet", framework.LabelCAPI, framework.La
 			template.Spec.Template.Spec.SpotMarketOptions = &awsv1.SpotMarketOptions{}
 		})
 	})
+
+	//OCP-84243 - [CAPI] AWS capacity reservation preference None should work correctly.
+	It("should be able to run a machine with capacity reservation preference None", func() {
+		awsMachineTemplate = newAWSMachineTemplate(awsMachineTemplateName, mapiDefaultProviderSpec)
+		awsMachineTemplate.Spec.Template.Spec.CapacityReservationPreference = awsv1.CapacityReservationPreferenceNone
+		Eventually(func() error {
+			return cl.Create(ctx, awsMachineTemplate)
+		}, framework.WaitShort, framework.RetryShort).Should(Succeed(), "Failed to create awsmachinetemplate")
+		machineSetParams = framework.UpdateCAPIMachineSetName("aws-machineset-84243-none", machineSetParams)
+		machineSet, err = framework.CreateCAPIMachineSet(ctx, cl, machineSetParams)
+		Expect(err).ToNot(HaveOccurred(), "Failed to create CAPI machineset")
+		framework.WaitForCAPIMachinesRunning(ctx, cl, machineSet.Name)
+
+		instance := getAWSInstanceConfig(ctx, cl, oc, machineSet)
+		Expect(instance.CapacityReservationSpecification.CapacityReservationPreference).To(Equal(ptr.To("none")), fmt.Sprintf("Instance should have capacity reservation preference set to '%s'", "none"))
+		Expect(instance.CapacityReservationId).Should(BeNil(), "Instance should not in any capacity because capacityReservationPreference is None")
```

Comment:
I think there is a typo here

Reply:
I wonder where the typo is, I didn't find it.